app/jobs/imo/invite_unsubscribed_patients.rb

```diff
@@ -0,0 +1,27 @@
+class Imo::InviteUnsubscribedPatients
+  include Sidekiq::Worker
+
+  sidekiq_options queue: :default
+
+  REINVITATION_BUFFER = 6.months.freeze
+
+  def perform
+    return unless Flipper.enabled?(:imo_messaging)
+
+    next_messaging_time = Communication.next_messaging_time
+    patients.each do |patient|
```

Comment:
`find_each` here, otherwise we will get all contactable patients the first time in loaded into memory. 🤯 

Also, we can `pluck(:id)`, as that is the only thing we need here...not the whole object.

Reply:
I should definitely pluck. The find_each comment is irrelevant if I pluck, right? Are you saying the scope would load all contactable patients into memory before the rest of the query narrows it down? Not sure I follow that part.

Reply:
It would load everything into memory after all those scopes are applied...which would still be tens of thousands of patients on prod I'm pretty sure.  So too much to load into memory.

👍  to pluck